src/Factories/DocumentFactory.php

```diff
@@ -242,12 +239,12 @@ protected function getValuesForField($object, $options): array
      * @param Document $doc Solr document
      * @param array $options Custom options
      * @param string $type Type of Solr field
-     * @param DBField|string $value Value(s) of the field
+     * @param DBField|string|null $value Value(s) of the field
      */
     protected function addToDoc($doc, $options, $type, $value): void
     {
         /* Solr requires dates in the form 1995-12-31T23:59:59Z, so we need to normalize to GMT */
-        if ($type === 'tdate' || $value instanceof DBDate) {
+        if (($value && $type === 'tdate') || $value instanceof DBDate) {
```

Comment:
Without this change, `null` will be changed to `1970-01-01T00:00:00Z`

Reply:
Good change, but needs testing.
Also, the value can be 0 or false on purpose. This could possibly be hardened in the future. Not a high priority.

I would think `if (!$value || (other_value_checks))` might be sufficient?

Reply:
True, but I would expect `0` and `false` values to already be corrected by the SilverStripe ORM/database before we reach this point?

Reply:
Not really. 0 is treated as false, same as null. My suggestion would be wrong too. A hard check would be better
`($value !== false || $value !== null)` (something something PHP casting)